app/javascript/lib/metrics-api.js

```diff
@@ -174,31 +181,51 @@ function logClickButton(target) {
 }
 
 /**
- * Get label elements for an input element
- *
+ * Get associated or nearest parent label text for an input element
+ * returns empty string if none
  * From https://stackoverflow.com/a/15061155
  */
-function getLabelsForElement(element) {
+export function getLabelTextForElement(element, excludeElementText) {
   if (metricsApiMock === true) {return []} // Needed for metrics-api.test.js
 
-  let labels
-  const id = element.id
+  let label = null
+  let labelText = ''
 
-  if (element.labels) {
-    return element.labels
+  if (element.labels && element.labels[0]) {
+    label = element.labels[0]
   }
-
-  if (id) {
-    labels = Array.from(document.querySelector(`label[for='${id}']`))
+  if (!label) {
+    const id = element.id
+    if (id) {
+      label = document.querySelector(`label[for='${id}']`)
+    }
+  }
+  if (!label) { // traverse parents looking for label
+    let parent = element
+    while (parent = parent.parentNode) {
```

Comment:
Are we sure this won't get stuck in an infinite loop?  I'm assuming yes, because at a certain point you'll reach the top-level node in the DOM and have nowhere to go...

Reply:
yup, that's the idea (and this structure was in our code before)